fuse_core/test/example_loss.h

```diff
@@ -0,0 +1,86 @@
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef FUSE_CORE_TEST_EXAMPLE_LOSS_H  // NOLINT{build/header_guard}
+#define FUSE_CORE_TEST_EXAMPLE_LOSS_H  // NOLINT{build/header_guard}
+
+#include <fuse_core/loss.h>
+#include <fuse_core/macros.h>
+#include <fuse_core/serialization.h>
+
+#include <boost/serialization/access.hpp>
+#include <boost/serialization/base_object.hpp>
+#include <boost/serialization/export.hpp>
+
+#include <string>
+
+
+/**
+ * @brief Dummy loss implementation for testing
+ */
+class ExampleLoss : public fuse_core::Loss
+{
+public:
+  FUSE_LOSS_DEFINITIONS(ExampleLoss);
+
+  explicit ExampleLoss(const double a = 1.0) : fuse_core::Loss(new ceres::HuberLoss(a)), a(a)
+  {
+  }
+
+  void initialize(const std::string& /*name*/) override {}
+
+  void print(std::ostream& /*stream = std::cout*/) const override {}
+
+  double a{ 1.0 };  // Public member variable just for testing
+
+private:
+  // Allow Boost Serialization access to private methods
+  friend class boost::serialization::access;
+
+  /**
+   * @brief The Boost Serialize method that serializes all of the data members in to/out of the archive
+   *
+   * @param[in/out] archive - The archive object that holds the serialized class members
+   * @param[in] version - The version of the archive being read/written. Generally unused.
+   */
+  template<class Archive>
+  void serialize(Archive& archive, const unsigned int /* version */)
+  {
+    archive & boost::serialization::base_object<fuse_core::Loss>(*this);
```

Comment:
Changed. There's no need to serialized it after my last changes because now the children implement the `lossFunction()` pure virtual method returning a `ceres::LossFunction` pointer that is built and allocated in the heap, as I think it was the original intend. Note this could be changed in a future PR to make it more efficient.

Example implementation for the `HuberLoss` children:
``` c++
ceres::LossFunction* HuberLoss::lossFunction() const
{
  return new ceres::HuberLoss(a_);
}
```